File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

// global variables
int letters = 0;
int words = 1;
int sentences = 0;

int main(void)
{
    // prompt user for input
    char *paragraph = get_string( "Text:");
    // calculate string length
    int length = (int) strlen(paragraph);

    // loop for length of paragraph
    for (int i = 0; i < length; i++)
    {
        // check if its a letter
        if ((paragraph[i] >= 65 && paragraph[i] <= 90) || (paragraph[i] >= 97 && paragraph[i] <= 122))
        {
            letters++;
        }
            // check if its a word
        else if (paragraph[i] == 32)
        {
            words++;
        }
            // check if its a sentence
        else if (paragraph[i] == 46 || paragraph[i] == 33 || paragraph[i] == 63)
        {
            sentences++;
        }
    }
    // calculate index components and then index, then round it to nearest int
    float L = (float) letters * 100 / (float) words;
    float S = (float) sentences * 100 / (float) words;
    int index = (int) round((0.0588 * L - 0.296 * S - 15.8));
    // printf("%d letters\n%d words\n%d sentences\nindex = %d\n", letters, words, sentences, index);
    // print output based on index
    if (index < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    else
    {
        printf("Grade %d\n", index);
        return 0;
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string
import math

# define global variables to be accessed by all functions
letters = 0
words = 0
sentences = 0


def main():
    # get string from user
    string = get_string("input string: ")
    count_fun(string)
    indexing_fun()


# counts all letters, words, and sentences in a paragraph
def count_fun(string):
    global letters
    global words
    global sentences
    
    # for each char in the string
    for i in string:
        # if its an alphabet increase letters count
        if i.isalpha():
            letters += 1
        # if its a space increase words count
        elif i == ' ':
            words += 1
        # if its a comma, a question mark, or an exclamation mark, increase sentences count
        elif i == '.' or i == '!' or i == '?':
            sentences += 1
    words += 1
    
# print output based on calculated index
def print_grade(index):
    if index < 1:
        print("Before Grade 1")
    elif index >= 1 and index < 16:
        print(f"Grade {index}")
    elif index >= 16:
        print('Grade 16+')

# calculate index using words, letters, and sentences counted, and then call the print_grade function
def indexing_fun():
    L = letters / words * 100
    S = sentences / words * 100
    
    index = round(round(0.0588 * L, 2) - round(0.296 * S, 2) - 15.8, 0)
    print_grade(index)


main()